### PR TITLE
test/e2e: Fix flaking subresource test

### LINKED ISSUE #116614: Kubernetes e2e suite: [It] [sig-cli] Kubectl client kubectl subresource flag GET on status subresource of built-in type (node) returns identical info as GET on the built-in type expand_less
### Which jobs are flaking?

Example: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/116611/pull-kubernetes-e2e-gce/1635729618159800320/

@MadhavJivrajani this PR seems to be flaking: https://github.com/kubernetes/kubernetes/pull/116590

comparing age of a node from the output of two commands: https://github.com/kubernetes/kubernetes/blob/9c2d28f7d55a5210d02b9a3c7dc6449b9e125340/test/e2e/kubectl/kubectl.go#L1989-L1993

### Which tests are flaking?

Kubernetes e2e suite: [It] [sig-cli] Kubectl client kubectl subresource flag GET on status subresource of built-in type (node) returns identical info as GET on the built-in type expand_less

### Since when has it been flaking?

5 hours back

### Testgrid link

_No response_

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig cli

### PR DESCRIPTION
**Pull request by MadhavJivrajani**
#### What type of PR is this?
/kind flake

#### What this PR does / why we need it:

Avoid comparing fields that might end up changing
between two invocations of kubectl.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/116614

#### Special notes for your reviewer:

I also thought of only comparing the headers, but thought this might be better. Happy to revise the approach.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource
```
/cc @aojea 
/assign @soltysh

### CODE REVIEW

**Comment by aojea** [comment]
/assign @seans3 

can you ptal? it maybe late for Maciej

Thanks for the quick fix

**Review by SergeyKanzhelev — APPROVED** [review]
straightforward change

/lgtm

File: test/e2e/kubectl/kubectl.go [hunk 1]
```diff
@@ -1990,7 +1990,22 @@ metadata:
...
+			requiredOutput := [][]string{
+				{"NAME"},
+				{"STATUS"},
+				{"ROLES"},
+				{"AGE"},
+				{"VERSION"},
+				{node.Name},                           // check for NAME
+				{""},                                  // avoid comparing STATUS
+				{""},                                  // avoid comparing ROLES
+				{""},                                  // avoid comparing AGE
```
Comment (liggitt) [inline test/e2e/kubectl/kubectl.go:2004]: these don't really make sense with what checkOutput does...

**Review by liggitt — COMMENTED** [review]
/approve
just to deflake master for code freeze, but I don't think empty strings passed to checkOutput make sense

a follow-up to make this output structured data with `-o jsonpath` would be more reliable and more correct

File: test/e2e/kubectl/kubectl.go (hunk 1 above)
Comment (MadhavJivrajani) [reply test/e2e/kubectl/kubectl.go:2004]: Yeah I agree it's a little hacky to workaround. I'll send a follow up

**Pull request merged**